contracts/token/erc20/preset/ImmutableERC20.sol

```diff
@@ -37,6 +40,7 @@ contract ImmutableERC20 is Ownable, ERC20Permit, MintingAccessControl {
      * @param amount  The amount of tokens to mint.
      */
     function mint(address to, uint256 amount) external onlyRole(MINTER_ROLE) {
+        require(totalSupply() + amount <= _maxSupply, "ImmutableERC20: max supply exceeded");
```

Comment:
should this come from errors file? https://github.com/immutable/contracts/blob/main/contracts/errors/Errors.sol